Gems/ROS2/Code/Source/RobotImporter/URDF/SensorsMaker.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include "UrdfParser.h"
+#include <AzCore/Component/ComponentBus.h>
+#include <AzCore/Component/EntityId.h>
+#include <AzCore/Outcome/Outcome.h>
+
+#include <sdf/sdf.hh>
+
+namespace ROS2
+{
+    //! Populates a given entity with all the contents of the <sensor> tag in robot description of Gazebo.
+    //! In SDFormat, sensors are specified as children of link of joint.
+    class SensorsMaker
+    {
+    public:
```

Comment:
Or Alternatively we could have the SensorsMaker constructor populate the `AZStd::vector<ROS2::SDFormat::SensorImporterHook>` and store it as a member or support passing in a pre-populated vector of SensorImporterHooks, to save on initialization time.

```c++
//! Store a user supplied list of importer hooks
//! The hooks will be used in AddSensors.
//! @param contiguous container of Sensor importer Hooks
SensorsMaker(AZStd::span<SDFormat::SensorImporterHook> sensorHooks);